src/server/dfly_main.cc

```diff
@@ -701,6 +702,43 @@ void PrintBasicUsageInfo() {
   std::cout << endl;
 }
 
+extern char** environ;
+
+void ParseFlagsFromEnv() {
+  const auto& flags = absl::GetAllFlags();
+  for (char** env = environ; *env != nullptr; env++) {
+    constexpr string_view kPrefix = "DFLY_";
+    string_view environ_var = *env;
+    if (absl::StartsWith(environ_var, kPrefix)) {
+      // Per 'man environ', environment variables are included with their values
+      // in the format "name=value". Need to strip them apart, in order to work with flags object
+      pair<string_view, string_view> environ_pair =
+          absl::StrSplit(absl::StripPrefix(environ_var, kPrefix), '=');
+      const auto& [flag_name, flag_value] = environ_pair;
+      const auto entry = flags.find(flag_name);
+      if (entry != flags.end()) {
+        if (absl::flags_internal::WasPresentOnCommandLine(flag_name)) {
+          // If already specified in the command line, then just ignore it.
+          // There's an alternative, non-internal method `IsSpecifiedOnCommandLine`
+          // within `flag` object, but it's deprecated.
```

Comment:
I didn't realize that, I'd actually vote for using that instead
I guess internal is worse than deprecated?
:disappointed:

Reply:
sorry, just tested it and it looked like a private method!